OlinkAnalyze/vignettes/bridging_introduction.Rmd

```diff
@@ -0,0 +1,233 @@
+        strip.text = element_text(size=16),
+        legend.title = element_blank(),
+        legend.position = "none")
+```  
+
+Use PCA plot to visualize sample-to-sample distance before bridging.
+
+```{r pca1, message=FALSE, fig.cap="PCA plot of combined datasets before bridging"}
+## before bridging
+
+### Generate unique SampleIDs
+npx_1 <- npx_data1 %>% 
+  mutate(dataset = "data1") %>% 
+  mutate(SampleID = paste0(dataset, PlateID, SampleID))
+  
+npx_2 <- npx_data2 %>% 
+  mutate(dataset = "data2") %>% 
+  mutate(SampleID = paste0(dataset, PlateID, SampleID))
+             
+npx_before_br <- rbind(npx_1, npx_2)
+
+### PCA plot
+OlinkAnalyze::olink_pca_plot(df = npx_before_br,
+                               color_g = "dataset", 
+                             byPanel = TRUE, 
+                             coloroption = c('orange', 'darkblue')) 
+
+```
+
+
+
+## Perform bridging between two data sets
+
+We can use `olink_normalization()` function to bridge two datasets. The bridging 
+procedure is to first calculate **the median of the paired NPX differences** per
+assay between the bridging samples as adjustment factor then use these adjustment 
+factors to adjust NPX values between two datasets. The output from 
+`olink_normalization()` function is a NPX table with adjusted NPX value in the 
+column `NPX`.
+
+```{r bridging, message=FALSE}
+# Find overlapping samples
+npx_1 <- npx_data1 %>% 
+  mutate(dataset = "data1")
+npx_2 <- npx_data2 %>% 
+  mutate(dataset = "data2")
+
+overlap_samples <- intersect(npx_1$SampleID, npx_2$SampleID) %>% 
+  data.frame() %>% 
+  filter(!str_detect(., 'CONTROL_SAMPLE')) %>% #Remove control samples
+  pull(.)
+# Perform Bridging normalization
+npx_br_data <- olink_normalization(df1 = npx_1, 
+                                   df2 = npx_2, 
+                                   overlapping_samples_df1 = overlap_samples,
+                                   df1_project_nr = '20200001',
+                                   df2_project_nr = '20200002',
+                                   reference_project = '20200001')
+glimpse(npx_br_data)
+
+```
+
+## Perform evaluation of bridging normalization
+
+
+First, check NPX distribution in datasets after bridging normalization.
+
+```{r densitybr, message=FALSE}
+# Plot NPX density after bridging normalization
+
+npx_br_data %>% 
+  mutate(Panel = gsub("Olink ", "", Panel)) %>% 
+  ggplot(aes(x = NPX, fill = Panel)) +
+  geom_density(scale = 1, alpha = 0.7) + 
+  facet_grid(dataset~Panel) +
+  olink_fill_discrete(coloroption = c('teal', 'lightblue')) +
+  set_plot_theme() + 
+  ggtitle("After bridging normalization: NPX distribution") +
+  theme(axis.title.x = element_blank(),
+        axis.title.y = element_blank(),
+        strip.text = element_text(size=16),
+        legend.title = element_blank(),
+        legend.position = "none")
+
+``` 
+
+Then summarize number of assays that have adjustment factors in certain ranges.
+
```

Comment:
Add comment on how we interpret adjustments factors? For example, what to think about when they are high

Reply:
Could you please add your comments on how we interpret adjustments factors here?

Reply:
I think what @nicola-prot wrote on line 249 in her edits are good: High adjustment factors can result from variations between projects, such as panel versions or technical modifications. Such assays can be visualized individually with violin plots and may warrant further investigation to confirm they are still comparable between projects.